### PR TITLE
Preparation for release 0.1.0

### PR DESCRIPTION
**Pull request by RMeli**
* Added `CHANGELOG.md`
* Added `CITATIONS.cff`
* Added `RELEASE_PROCEDURE.md`
* Added `scripts/roll_release.sh`

The `roll_release.sh` script (from DLA-Future https://github.com/eth-cscs/DLA-Future/pull/1127) uses the official [GitHub CLI `gh`](https://cli.github.com) instead of [`hub`](https://hub.github.com) (untested!), and adds a check to `CITATIONS.cff` (https://github.com/eth-cscs/DLA-Future/pull/1141).

The authors added to `CITATIONS.cff` are the current `dla-future` and `pika` developers plus @mtaillefumier, in alphabetical order. Please let me know if the author list should be different.

### CODE REVIEW

**Comment by RMeli** [comment]
Blocked by https://github.com/eth-cscs/DLA-Future-Fortran/pull/3.

File: RELEASE_PROCEDURE.md
Comment (rasolca) [inline RELEASE_PROCEDURE.md:28]: Please note that you didn't use `###` in the spack package...

File: RELEASE_PROCEDURE.md
Comment (rasolca) [inline RELEASE_PROCEDURE.md:24]: path incorrect 😉

File: RELEASE_PROCEDURE.md
Comment (rasolca) [inline RELEASE_PROCEDURE.md:19]: Empty link for GitHub CLI

**Review by rasolca — COMMENTED** [review]
Quick question, do you want to be the only maintainer of the spack package? I'd add some extra names as well.

**Comment by RMeli** [comment]
Good point! I'm happy to add whoever wants to be added, of course.

To make things simple, just like this message if you want to be added as maintainer of the Spack package.

File: CITATION.cff
Comment (RMeli) [inline CITATION.cff:31]: I'm not entirely sure what should go here before doing an actual release.

Comment (msimberg) [reply CITATION.cff:31]: I suspect that this has to be left out from the first release. The field is optional anyway, and then you could fill in the concept DOI (and/or the versioned DOI) after 0.1.0 is out? As far as I understand it this field won't make a difference for the zenodo release anyway, but I may be wrong.

File: CITATION.cff
Comment (msimberg) [inline CITATION.cff:45]: Don't forget to change the date before the release (or after the release?). Maybe this could also be in the release script?

File: scripts/roll_release.sh
Comment (msimberg) [inline scripts/roll_release.sh:31]:
```suggestion
# Major and minor releases are made directly from main. Patch releases are branched out from the major
```

File: CITATION.cff
Comment (RMeli) [reply CITATION.cff:45]: I think before is fine? Great idea to add this check too to the release script!

File: CITATION.cff
Comment (RMeli) [reply CITATION.cff:31]: https://github.com/eth-cscs/DLA-Future-Fortran/issues/4

**Comment by RMeli** [comment]
cscs-ci run

**Review by rasolca — APPROVED** [review]

**Review by msimberg — APPROVED** [review]

**Review by mtaillefumier — APPROVED** [review]

**Comment by RMeli** [comment]
cscs-ci run

File: CITATION.cff
Comment (RMeli) [reply CITATION.cff]: I'll go with last name ordering since it's more standard.

**Pull request merged**
